### PR TITLE
Feature: Interface for database

### PR DESCRIPTION
**Pull request by TOPOFGR**
Details:

- Added interface to choose between Firestore and Realtime Database

- Added Firestore cloud functions

- Refactor for `setupProject.js`

### CODE REVIEW

File: firestore.rules [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+    }
+
+    function isAdmin() {
+      return request.auth.token.isAdmin == true;
+    }
+    
+
+    match /users/{userId} {
+      
+    function isUser() {
```
Comment (tpiaggio) [inline firestore.rules:16]: can we indent this function properly, use ES6 syntax and name it `isIdentified` ?

File: functions/setupProject.js [hunk 2]
```diff
@@ -1,54 +1,123 @@
...
+    type: 'password',
+    name: 'password',
+    message: 'Enter user password: ',
+    mask: '*',
+  },
+  {
+    type: 'list',
+    name: 'database',
+    message: 'Select the database of your choice:',
+    choices: ['Real Time Database', 'Firestore'],
```
Comment (tpiaggio) [inline functions/setupProject.js:36]: "Realtime", not "Real Time"

File: src/state/api/firestore.js [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+
+export const deleteDocument = (collection, id) => {
+  return getFirestoreRef(collection).doc(id).delete();
+};
+
+export const createDocument = (collection, id, values) => {
+  return getFirestoreRef(collection).doc(id).set(values);
+};
+
+export const modifyDocument = (collection, id, values) => {
```
Comment (tpiaggio) [inline src/state/api/firestore.js:47]: could we name this `updateDocument` instead?

File: README.md [hunk 4]
```diff
@@ -64,6 +64,7 @@ React Firebase Admin is our in-house admin dashboard boilerplate, used in many o
 - PWA ready thanks to CRA and Firebase
 - Multi-tenancy
 - Internationalization (English/Spanish)
+- Interface to choose between Real Time Database and Firestore
```
Comment (MateoKruk) [inline README.md:67]: Maybe we should change this to: "Ability to choose between realtime database or firestore"

File: functions/package.json [hunk 5]
```diff
@@ -20,7 +20,8 @@
     "firebase-admin": "^9.1.0",
     "firebase-function-tools": "^1.1.3",
     "firebase-functions": "^3.9.1",
-    "glob": "^7.1.6"
+    "glob": "^7.1.6",
+    "inquirer": "^7.3.3"
```
Comment (MateoKruk) [inline functions/package.json:24]: This package should be a dev dependency, right?

File: functions/test/db/users/onUpdate.test.ts [hunk 6]
```diff
@@ -3,16 +3,16 @@ import * as chai from 'chai';
 import onUpdate from '../../../src/db/users/onUpdate.function';
 import 'mocha';
 
-describe('onUpdate', () => {
+describe('onUpdate Real Time Database', () => {
```
Comment (MateoKruk) [inline functions/test/db/users/onUpdate.test.ts:6]: Realtime*

File: src/state/api/firestore.js [hunk 7]
```diff
@@ -0,0 +1,49 @@
+// eslint-disable-next-line no-unused-vars
```
Comment (MateoKruk) [inline src/state/api/firestore.js:1]: Why we do this rule override?

File: functions/package.json (hunk 5 above)
Comment (TOPOFGR) [reply functions/package.json:24]: Thought the same, but inquirer must be a dependency. Someone got the same issue [here](https://github.com/plopjs/node-plop/issues/117).

File: README.md [hunk 8]
```diff
@@ -64,6 +64,7 @@ React Firebase Admin is our in-house admin dashboard boilerplate, used in many o
 - PWA ready thanks to CRA and Firebase
 - Multi-tenancy
 - Internationalization (English/Spanish)
+- Ability to choose between realtime database or firestore
```
Comment (tpiaggio) [inline README.md:67]: Ability to choose between Firebase Realtime Database or Firestore

File: functions/setupProject.js [hunk 9]
```diff
@@ -1,54 +1,152 @@
...
+    name: 'password',
+    message: 'Enter user password: ',
+    mask: '*',
+  },
+  {
+    type: 'list',
+    name: 'database',
+    message: 'Select the database of your choice:',
+    choices: ['Realtime Database', 'Firestore'],
+  },
```
Comment (tpiaggio) [inline functions/setupProject.js:40]: Let's add a new question asking if we want to delete either 'Realtime Database' or 'Firestore' cloud functions. If the user selects 'No', then we're not going to delete the cloud functions, but if they select 'Yes', we delete unused cloud functions.

**Review by MateoKruk — APPROVED** [review]

**Pull request merged**